addon/lib/metrics.js

```diff
@@ -121,13 +151,32 @@ module.exports = {
     this.updateExperiment(addonId, {features: features});
   },
 
+  onReceiveVariantDefs: function(ev) {
+    if (!store.experimentVariants) {
+      store.experimentVariants = {};
+    }
+
+    const { subject, data } = ev;
+    const dataParsed = variantMaker.parseTests(JSON.parse(data));
+
+    store.experimentVariants[subject] = dataParsed;
+    Events.emit(EVENT_SEND_VARIANTS, {
+      data: JSON.stringify(dataParsed),
+      subject: self.id
+    });
+  },
+
   onExperimentPing: function(ev) {
     const { subject, data } = ev;
     const dataParsed = JSON.parse(data);
 
     // TODO: Map add-on ID (subject) to other pingTypes as necessary
     const pingType = 'testpilottest';
 
+    if (subject in store.experimentVariants) {
```

Comment:
+ quoting the metrics readme:

> The second payload's schema will be defined by each experiment and needs to remain flexible for rapid changes.

I read this to say that, as long as this happens:

> Please note that each experiment will need to define its payload schema before being able to record data.

We're able to go freeform with the payload as far as the Test Pilot addon is concerned, yeah?